htmresearch/frameworks/layers/l2_l4_inference.py

```diff
@@ -79,33 +79,105 @@
 
 """
 
-import os
-import random
 import collections
-import inspect
 import cPickle
+import inspect
+import os
+import random
 import matplotlib.pyplot as plt
 from tabulate import tabulate
 
 from htmresearch.support.register_regions import registerAllResearchRegions
 from htmresearch.frameworks.layers.laminar_network import createNetwork
 
 
+
+class LoggingDecorator(object):
```

Comment:
Is it possible to move it to htmresearch/support?  I would also like to use this decorator in the new class I am writing for L2456 networks?